src/Queue/index.test.ts

```diff
@@ -56,3 +56,23 @@ describe('ExpirableQueue', () => {
     expect(queue.size).toBe(0);
   });
 });
+
+describe('ExpirableQueue hooks', () => {
+  it('should call the beforeExpire hook before expiring an entry', () => {
+    const queue = new ExpirableQueue([1, 2, 3], { defaultTtl: 10 });
+    const beforeExpire = jest.fn();
+    queue.addHook('beforeExpire', beforeExpire);
```

Comment:
You can check if the item is not yet expired.

Reply:
You are right, done!